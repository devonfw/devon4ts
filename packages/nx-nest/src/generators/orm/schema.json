{
  "$schema": "http://json-schema.org/schema",
  "$id": "InitTypeormSchematic",
  "title": "Init Typeorm Schema",
  "type": "object",
  "properties": {
    "orm": {
      "type": "string",
      "enum": ["typeorm", "prisma", "drizzle"],
      "default": "typeorm",
      "$default": {
        "$source": "argv",
        "index": 1
      },
      "x-prompt": "Which ORM do you want to use?"
    },
    "db": {
      "type": "string",
      "enum": ["postgres", "cockroachdb", "mariadb", "mysql", "sqlite", "oracle", "mssql", "mongodb"],
      "default": "sqlite",
      "$default": {
        "$source": "argv",
        "index": 2
      },
      "x-prompt": "Which database engine do you want to use?"
    },
    "projectName": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "In which project would you like to initialize the ORM?"
    }
  },
  "required": ["projectName"]
}
