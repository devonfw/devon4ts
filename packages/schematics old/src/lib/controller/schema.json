{
  "$schema": "http://json-schema.org/schema",
  "$id": "ControllerSchema",
  "title": "Controller Schema",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the controller.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the controller?"
    },
    "spec": {
      "default": true,
      "type": "boolean",
      "description": "Specifies if a spec file is generated."
    }
  },
  "required": [
    "name"
  ]
}
