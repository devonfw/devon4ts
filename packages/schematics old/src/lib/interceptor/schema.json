{
  "$schema": "http://json-schema.org/schema",
  "$id": "SchematicsNestInterceptor",
  "title": "Nest Interceptor Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the interceptor.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the interceptor?"
    },
    "spec": {
      "default": true,
      "description": "Specifies if a spec file is generated."
    }
  },
  "required": ["name"]
}
