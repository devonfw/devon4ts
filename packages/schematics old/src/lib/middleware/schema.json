{
  "$schema": "http://json-schema.org/schema",
  "$id": "SchematicsNestMiddleware",
  "title": "Nest Middleware Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the middleware.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the middleware?"
    },
    "spec": {
      "default": true,
      "description": "Specifies if a spec file is generated."
    }
  },
  "required": ["name"]
}
