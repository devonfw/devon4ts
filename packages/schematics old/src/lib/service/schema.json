{
  "$schema": "http://json-schema.org/schema",
  "$id": "ServiceSchematic",
  "title": "Service Schema",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the service.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the service?"
    },
    "spec": {
      "default": true,
      "description": "Specifies if a spec file is generated."
    }
  },
  "required": [
    "name"
  ]
}
