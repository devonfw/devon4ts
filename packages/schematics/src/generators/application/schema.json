{
  "$schema": "https://json-schema.org/schema",
  "$id": "Devon4tsNestApplicationGenerator",
  "title": "devon4ts Application Options Schema",
  "description": "devon4ts Application Options Schema.",
  "cli": "nx",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the application.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the node application?",
      "pattern": "^[a-zA-Z][^:]*$"
    },
    "directory": {
      "description": "The directory of the new application.",
      "type": "string"
    },
    "projectNameAndRootFormat": {
      "description": "Whether to generate the project name and root directory as provided (`as-provided`) or generate them composing their values and taking the configured layout into account (`derived`).",
      "type": "string",
      "enum": ["as-provided", "derived"]
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    },
    "skipPackageJson": {
      "description": "Do not add dependencies to `package.json`.",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    },
    "unitTestRunner": {
      "description": "Test runner to use for unit tests.",
      "type": "string",
      "enum": ["jest", "none"],
      "default": "jest"
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["jest", "none"],
      "description": "Test runner to use for end to end (e2e) tests",
      "default": "jest"
    },
    "tags": {
      "description": "Add tags to the application (used for linting).",
      "type": "string"
    },
    "frontendProject": {
      "description": "Frontend project that needs to access this application. This sets up proxy configuration.",
      "type": "string",
      "x-priority": "important"
    },
    "standaloneConfig": {
      "description": "Split the project configuration into `<projectRoot>/project.json` rather than including it inside `workspace.json`.",
      "type": "boolean",
      "default": true,
      "x-deprecated": "Nx only supports standaloneConfig"
    },
    "setParserOptionsProject": {
      "type": "boolean",
      "description": "Whether or not to configure the ESLint `parserOptions.project` option. We do not do this by default for lint performance reasons.",
      "default": false
    },
    "strict": {
      "type": "boolean",
      "description": "Adds strictNullChecks, noImplicitAny, strictBindCallApply, forceConsistentCasingInFileNames and noFallthroughCasesInSwitch to tsconfig.",
      "default": false
    }
  },
  "additionalProperties": false,
  "required": ["name"]
}
