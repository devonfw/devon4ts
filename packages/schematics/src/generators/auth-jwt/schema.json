{
  "$schema": "https://json-schema.org/schema",
  "$id": "AuthJwt",
  "title": "",
  "type": "object",
  "properties": {
    "projectName": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    }
  },
  "required": ["projectName"]
}
