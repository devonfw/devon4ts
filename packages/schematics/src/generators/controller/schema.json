{
  "$schema": "http://json-schema.org/schema",
  "$id": "ControllerSchema",
  "title": "Controller Schema",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the controller.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the controller?"
    },
    "project": {
      "type": "string",
      "description": "Target application where the controller is going to be created",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    }
  },
  "required": ["name", "project"]
}
