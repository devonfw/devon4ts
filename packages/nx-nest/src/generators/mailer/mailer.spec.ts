import { Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import applicationGenerator from '../application/application';
import convictGenerator from '../convict/convict';
import { mailerGenerator } from './mailer';
import { MailerGeneratorSchema } from './schema';

describe('mailer generator', () => {
  describe('mailer generator without convict', () => {
    let tree: Tree;
    const options: MailerGeneratorSchema = { projectName: 'test' };
    beforeAll(async () => {
      tree = createTreeWithEmptyWorkspace();
      await applicationGenerator(tree, {
        name: options.projectName,
        projectNameAndRootFormat: 'as-provided',
        directory: 'apps/test',
      });
      await mailerGenerator(tree, options);
    }, 60000);

    it('should add dependencies to package.json', async () => {
      const fileContent = tree.read('package.json')?.toString('utf-8');
      expect(fileContent).toMatch(/"dependencies": {(.|\n)*"handlebars":/g);
      expect(fileContent).toMatch(/"dependencies": {(.|\n)*"@devon4ts\/mailer":/g);
    });

    it('should generate the template files', () => {
      expect(
        tree.exists(`packages/nx-nest/apps/${options.projectName}/src/assets/templates/partials/layout.handlebars`),
      ).toBeTruthy();
      expect(
        tree.exists(`packages/nx-nest/apps/${options.projectName}/src/assets/templates/views/example.handlebars`),
      ).toBeTruthy();
    });

    it('should add MailerModule to core.module', async () => {
      const fileContent = tree
        .read(`./packages/nx-nest/apps/${options.projectName}/src/app/core/core.module.ts`)
        ?.toString('utf-8');

      expect(fileContent).toMatchSnapshot();
    });

    it('should merge the compose.yaml', () => {
      expect(tree.read('compose.yaml')?.toString('utf-8')).toMatchSnapshot();
    });
  });

  describe('mailer generator with convict', () => {
    let tree: Tree;
    const options: MailerGeneratorSchema = { projectName: 'test' };
    beforeAll(async () => {
      tree = createTreeWithEmptyWorkspace();
      await applicationGenerator(tree, {
        name: options.projectName,
        projectNameAndRootFormat: 'as-provided',
        directory: 'apps/test',
      });
      tree.write(
        'compose.yaml',
        `version: '3'
services:
  test:
    image: test`,
      );
      await convictGenerator(tree, options);
      await mailerGenerator(tree, options);
    }, 60000);

    it('should add dependencies to package.json', async () => {
      const fileContent = tree.read('package.json')?.toString('utf-8');
      expect(fileContent).toMatch(/"dependencies": {(.|\n)*"handlebars":/g);
      expect(fileContent).toMatch(/"dependencies": {(.|\n)*"@devon4ts\/mailer":/g);
    });

    it('should generate the template files', () => {
      expect(
        tree.exists(`packages/nx-nest/apps/${options.projectName}/src/assets/templates/partials/layout.handlebars`),
      ).toBeTruthy();
      expect(
        tree.exists(`packages/nx-nest/apps/${options.projectName}/src/assets/templates/views/example.handlebars`),
      ).toBeTruthy();
    });

    it('should add MailerModule to core.module', async () => {
      const fileContent = tree
        .read(`./packages/nx-nest/apps/${options.projectName}/src/app/core/core.module.ts`)
        ?.toString('utf-8');
      expect(fileContent).toContain('MailerModule');
      expect(fileContent).toMatchSnapshot();
    });

    it('should add mailer configuration src/app/shared/app-config.ts', async () => {
      const fileContent = tree
        .read(`./packages/nx-nest/apps/${options.projectName}/src/app/app-config.ts`)
        ?.toString('utf-8');
      expect(fileContent).toContain('mailer');
      expect(fileContent).toMatchSnapshot();
    });

    it('should add mailer configuration src/config.ts', async () => {
      const fileContent = tree.read(`./packages/nx-nest/apps/${options.projectName}/src/config.ts`)?.toString('utf-8');
      expect(fileContent).toContain('mailer');
      expect(fileContent).toMatchSnapshot();
    });

    it('should merge the compose.yaml', () => {
      expect(tree.read('compose.yaml')?.toString('utf-8')).toMatchSnapshot();
    });
  });
});
