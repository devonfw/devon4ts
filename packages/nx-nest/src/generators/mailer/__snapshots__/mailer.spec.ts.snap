// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mailer generator mailer generator with convict should add MailerModule to core.module 1`] = `
"import { ClassSerializerInterceptor, Global, Module } from '@nestjs/common';
import { LoggerModule } from '@proj/shared/logger';
import { APP_INTERCEPTOR } from '@nestjs/core';
import { CONFIG_PROVIDER } from '@proj/shared/config';
import config from '../../config';
import { MailerModule } from '@devon4ts/mailer';
import { join } from 'path';

@Global()
@Module({
  imports: [ MailerModule.register({
      mailOptions: {
        host: config.mailer.mailOptions.host,
        port: config.mailer.mailOptions.port,
        secure: config.mailer.mailOptions.secure,
        tls: {
          rejectUnauthorized: config.mailer.mailOptions.tlsRejectUnauthorized,
        },
      },
      emailFrom: config.mailer.emailFrom,
      hbsOptions: {
        templatesDir: join(__dirname, '../..', config.mailer.hbsOptions.templatesDir),
        partialsDir: join(__dirname, '../..', config.mailer.hbsOptions.partialsDir),
        helpers: [],
      },
    }),LoggerModule],
  controllers: [],
  providers: [
    {
      provide: CONFIG_PROVIDER,
      useValue: config,
    },
    { provide: APP_INTERCEPTOR, useClass: ClassSerializerInterceptor },
  ],
  exports: [ MailerModule,CONFIG_PROVIDER, LoggerModule],
})
export class CoreModule {}
"
`;

exports[`mailer generator mailer generator with convict should add mailer configuration src/app/shared/app-config.ts 1`] = `
"import { BaseConfig } from '@proj/shared/config';

export interface AppConfig extends BaseConfig {
  mailer: {
        mailOptions: {
          host: string;
          port: number;
          secure: boolean;
          tlsRejectUnauthorized: boolean;
        };
        emailFrom: string;
        hbsOptions: {
          templatesDir: string;
          partialsDir: string;
        };
      };
}
"
`;

exports[`mailer generator mailer generator with convict should add mailer configuration src/config.ts 1`] = `
"import { config, defaultConfig } from '@proj/shared/config';
import { Schema } from 'convict';
import { AppConfig } from './app/app-config';

export default config({
  ...defaultConfig,
  mailer: {
      mailOptions: {
        host: {
          doc: 'Mail server host URL',
          default: 'localhost',
          format: String,
          env: 'MAILER_HOST',
        },
        port: {
          doc: 'Mail server port',
          default: 1025,
          format: 'port',
          env: 'MAILER_PORT',
        },
        secure: {
          doc: 'Is the mailer server secured?',
          default: false,
          format: Boolean,
          env: 'MAILER_SECURE',
        },
        tlsRejectUnauthorized: {
          doc: 'Reject unauthorized TLS connections?',
          default: false,
          format: Boolean,
          env: 'MAILER_TLS_REJECT_UNAUTHORIZED',
        },
      },
      emailFrom: {
        doc: 'Email that will be used as sender',
        default: 'noreply@example.com',
        format: String,
        env: 'MAILER_EMAIL_FROM',
        arg: 'emailFrom',
      },
      hbsOptions: {
        templatesDir: {
          doc: 'Relative path to handlebars views folder',
          default: 'assets/templates/views',
          format: String,
        },
        partialsDir: {
          doc: 'Relative path to handlebars partials folder',
          default: 'assets/templates/partials',
          format: String,
        },
      },
    }
} as Schema<AppConfig>);
"
`;

exports[`mailer generator mailer generator with convict should merge the compose.yaml 1`] = `
"version: '3'
services:
  test:
    image: test
  maildev:
    image: djfarrelly/maildev
    ports:
      - '1025:25'
      - '1080:80'
"
`;

exports[`mailer generator mailer generator without convict should add MailerModule to core.module 1`] = `
"import { ClassSerializerInterceptor, Global, Module } from '@nestjs/common';
import { LoggerModule } from '@proj/shared/logger';
import { APP_INTERCEPTOR } from '@nestjs/core';
import { MailerModule } from '@devon4ts/mailer';
import { join } from 'path';

@Global()
@Module({
  imports: [ MailerModule.register({
      mailOptions: {
        host: 'localhost',
        port: 1025,
        secure: false,
        tls: {
          rejectUnauthorized: false,
        },
      },
      emailFrom: 'noreply@example.com',
      hbsOptions: {
        templatesDir: join(__dirname, '../..', 'assets/templates/views'),
        partialsDir: join(__dirname, '../..', 'assets/templates/partials'),
        helpers: [],
      },
    }),LoggerModule],
  controllers: [],
  providers: [{ provide: APP_INTERCEPTOR, useClass: ClassSerializerInterceptor }],
  exports: [ MailerModule,LoggerModule],
})
export class CoreModule {}
"
`;

exports[`mailer generator mailer generator without convict should merge the compose.yaml 1`] = `
"version: '3'
services:
  maildev:
    image: djfarrelly/maildev
    ports:
      - '1025:25'
      - '1080:80'
"
`;
